case-studies/antipatterns/classic/app/pods/side-effects/controller.js

```diff
@@ -1,23 +1,24 @@
-import Controller from '@ember/controller';
-import { computed } from '@ember/object';
+import Controller from "@ember/controller";
+import { computed } from "@ember/object";
 
 export default Controller.extend({
-  firstName: 'John',
-  lastName: 'Doe',
-
-  fullNameLength: null,
-
-  fullName: computed('firstName', 'lastName', function () {
-    const fullName = `${this.get('firstName')} ${this.get('lastName')}`;
-
-    this.set('fullNameLength', fullName.length);
+  firstName: "John",
+  lastName: "Doe",
 
+  fullName: computed("firstName", "lastName", function () {
+    const fullName = `${this.get("firstName")} ${this.get("lastName")}`;
     return fullName;
   }),
 
+  fullNameLength: computed("fullName", function () {
+    if (this.get("fullName")) {
+      return this.get("fullName.length");
+    } else return null;
+  }),
+
   actions: {
     change(property, event) {
       this.set(property, event.target.value);
-    }
+    },
   },
 });
```

Comment:
Unrelated file included into the PR. It's OK in the exercise, but it would not be acceptable in a real app.